__tests__/app.test.js

```diff
@@ -169,3 +169,61 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+describe("POST /api/articles/:article_id/comments", () => {
+  test("creates and adds a new comment with username and body to the database", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "Superb article, thumbs up from me!",
+    };
+    return request(app)
+      .post("/api/articles/9/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body: { comment } }) => {
+        expect(comment).toEqual(
+          expect.objectContaining({
+            article_id: 9,
+            author: "butter_bridge",
+            body: "Superb article, thumbs up from me!",
+            votes: 0,
+          })
+        );
+      });
+  });
+  test("does not add a new comment when given an invalid artile id", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "Superb article, thumbs up from me!",
+    };
+    return request(app)
+      .post("/api/articles/99/comments")
+      .send(newComment)
+      .expect(404)
+      .then((response) => {
+        const comment = response.body.comment;
+        const msg = response.body.msg;
+        expect(comment).toEqual(
+          expect.not.objectContaining({
+            author: "butter_bridge",
+            body: "Superb article, thumbs up from me!",
+          })
+        )
+        expect(msg).toBe("No article found for article_id 99");
+      });
+  });
+
+  test("returns a 400 bad request if data type passed for article number is NaN whilst POSTING a new comment", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "Superb article, thumbs up from me!",
+    };
+    return request(app)
+      .post("/api/articles/banana/comments")
+      .send(newComment)
+      .expect(400)
+      .then((response) => {
+        const msg = response.body.msg;
+        expect(msg).toBe("Bad Request");
+      });
+  });
+});
```

Comment:
Nice start with the testing, could we also just add:
- Another 201 that ignores unnecessary properties (i.e. if you put something else apart from username and body in the newComment)
- A 400 for if the newComment is missing the username or body properties
- A 404 for if the username does not exist.